### PR TITLE
USD to SDF: Added visuals

### PR DESCRIPTION
**Pull request by ahcorde**
# 🎉 New feature

## Summary

Added visuals with materials

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:153]:
```suggestion
std::string directoryFromUSDPath(const std::string &_primPath)
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:172]:
```suggestion
/// \param[in] _prim Reference to the USD prim
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:185]:
```suggestion
  for (const auto &p : paths)
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:200]:
```suggestion
/// \param[in] _prim Reference to the USD prim
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:215]: I don't see `data` being used in this method, can these lines be removed?

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:308]:
```suggestion
/// \param[in] _prim Reference to the USD prim
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:353]:
```suggestion
  for (const auto &textCoord : textCoords)
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:358]:
```suggestion
  for (const auto &point : points)
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:365]:
```suggestion
  for (const auto &normal : normals)
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:409]: can this line be deleted?

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:424]: can this line be deleted?

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:440]:
```suggestion
/// \param[in] _prim Reference to the USD prim
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:447]: nit: I don't believe we use `const` for primitive types

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:450]: Should this variable be renamed to `variant_cube`, since this method parses a USD cube?

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:446]:
```suggestion
  const ignition::math::Vector3d &_scale,
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:467]:
```suggestion
/// \param[in] _prim Reference to the USD prim
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:473]:
```suggestion
  const ignition::math::Vector3d &_scale,
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:474]: nit: I don't believe we use `const` for primitive types

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:488]:
```suggestion
/// \param[in] _prim Reference to the USD prim
```

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:496]: nit: I don't believe we use `const` for primitive types

File: usd/src/usd_parser/USDLinks.cc
Comment (adlarkin) [inline usd/src/usd_parser/USDLinks.cc:572]: Can we delete this code? Taking a closer look at it, it seems to define `tmpPrim`, which is a local variable that I don't see being used anywhere else. So, I don't see why this is needed.

**Review by adlarkin — COMMENTED** [review]
I made a few updates in https://github.com/ignitionrobotics/sdformat/pull/897/commits/b0085d0b620429c95ca89f14990160a6306ef916. Overall, LGTM, but I do have one last question before approving/merging.

File: usd/src/usd_parser/USDLinks.cc
Comment (ahcorde) [reply usd/src/usd_parser/USDLinks.cc:572]: nope, I made this fix https://github.com/ignitionrobotics/sdformat/pull/897/commits/d8de8ba2e9bfb182ac3a7104cb300f5431372fdf

Nice catch

**Review by adlarkin — APPROVED** [review]
LGTM as long as CI comes back green :+1:

**Review by adlarkin — APPROVED** [review]

**Pull request merged**
